packages/bsky/src/auth-verifier.ts

```diff
@@ -134,12 +134,29 @@ export class AuthVerifier {
           iss: null,
           aud: null,
         })
-        if (!opts.skipAudCheck && !this.standardAudienceDids.has(aud)) {
+
+        const [audDid, audServiceId] = aud.split('#')
+
+        if (!opts.skipAudCheck && !this.standardAudienceDids.has(audDid)) {
           throw new AuthRequiredError(
             'jwt audience does not match service did',
             'BadJwtAudience',
           )
         }
+
+        const hasServiceId = aud.includes('#')
```

Comment:
It seems that in parsing out the service id we've already done the work to know whether the audience "has a service id".
```suggestion
        const hasServiceId = audServiceId !== undefined
```